ingestion/src/metadata/ingestion/source/messaging/kafka.py

```diff
@@ -203,31 +199,31 @@ def _parse_topic_metadata(self, topic: str) -> Optional[Schema]:
         return schema
 
     def _get_sample_data(self, topic_name):
+        sample_data = []
         try:
             self.consumer_client.subscribe([topic_name.__root__])
-            registry_client = SchemaRegistry(
-                self.service_connection.schemaRegistryURL,
-                headers={"Content-Type": "application/vnd.schemaregistry.v1+json"},
+            logger.info(
+                f"Kafka consumer polling for sample messages in topic {topic_name.__root__}"
             )
-            avro_serde = AvroKeyValueSerde(registry_client, topic_name.__root__)
-            logger.info("Kafka consumer polling for sample messages")
             messages = self.consumer_client.consume(num_messages=10, timeout=10)
-            sample_data = []
-            if len(messages) > 0:
-                for message in messages:
-                    sample_data.append(
-                        str(avro_serde.value.deserialize(message.value()))
-                    )
-            topic_sample_data = TopicSampleData(messages=sample_data)
-            self.consumer_client.unsubscribe()
-            return topic_sample_data
         except Exception as e:
             logger.error(
                 f"Failed to fetch sample data from topic {topic_name.__root__}"
             )
             logger.error(traceback.format_exc())
             logger.error(sys.exc_info()[2])
-        return None
+        else:
+            if messages:
+                for message in messages:
+                    sample_data.append(
+                        str(
+                            self.consumer_client._serializer.decode_message(
```

Comment:
Not entirely sure but is there an issue with accessing a protected member (_serializer) of the consumer_client class directly?

Reply:
hmm. When I tried it works fine. Yes its protected member shows as warning in the editor. But it works during the runtime. Unfortunately python client from confluent doesn't do a good job of providing good API here.